samples/core/guide/autograph_control_flow.ipynb

```diff
@@ -0,0 +1,770 @@
+      "metadata": {
+        "id": "-pkEH6OecW7h",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### Assert\n",
+        "\n",
+        "Let's try some other useful Python constructs, like `print` and `assert`. We automatically convert Python `assert` statements into the equivalent `tf.Assert` code.  "
+      ]
+    },
+    {
+      "metadata": {
+        "id": "IAOgh62zCPZ4",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "def f(x):\n",
+        "  assert x != 0, 'Do not pass zero!'\n",
+        "  return x * x\n",
+        "\n",
+        "tf_f = autograph.to_graph(f)\n",
+        "\n",
+        "with tf.Graph().as_default():  \n",
+        "  with tf.Session():\n",
+        "    try:\n",
+        "      print(tf_f(tf.constant(0)).eval())\n",
+        "    except tf.errors.InvalidArgumentError as e:\n",
+        "      print('Got error message:\\n%s' % e.message)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "KRu8iIPBCQr5",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### Print\n",
+        "\n",
+        "You can also use plain Python `print` functions in in-graph"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "ySTsuxnqCTQi",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "@autograph.convert()\n",
+        "def f(n):\n",
+        "  if n >= 0:\n",
+        "    while n < 5:\n",
+        "      n += 1\n",
+        "      print(n)\n",
+        "  return n\n",
+        "    \n",
+        "with tf.Graph().as_default():\n",
+        "  with tf.Session():\n",
+        "    f(tf.constant(0)).eval()"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "NqF0GT-VCVFh",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### Lists\n",
+        "\n",
+        "Appending to lists in loops also works (we create a `TensorArray` for you behind the scenes)"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "ABX070KwCczR",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "def f(n):\n",
+        "  z = []\n",
+        "  # We ask you to tell us the element dtype of the list\n",
+        "  z = autograph.utils.set_element_type(z, tf.int32)\n",
+        "  for i in range(n):\n",
+        "    z.append(i)\n",
+        "  # when you're done with the list, stack it\n",
+        "  # (this is just like np.stack)\n",
+        "  return autograph.stack(z) \n",
+        "\n",
+        "tf_f = autograph.to_graph(f)\n",
+        "\n",
+        "with tf.Graph().as_default():  \n",
+        "  with tf.Session():\n",
+        "    print(tf_f(tf.constant(3)).eval())"
```

Comment:
Any thought about using eval() / sess.run() consistently? I'm ok if you feel both should be exercised.

Reply:
> Any thought about using eval() / sess.run() consistently? I'm ok if you feel both should be exercised.

Consistency, sure! Do you have a preferred style? I have no strong feelings on this.

Reply:
I generally use sess.run(), because eval() is not as idiomatic.